Migrate GPUProcess structures to generated serialization

achristensen07 opened the pull request:
#### 4efc1a51c567db867f6f90be7e5da37e267be828
<pre>
Migrate GPUProcess structures to generated serialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=247150">https://bugs.webkit.org/show_bug.cgi?id=247150</a>
rdar://101651187

Reviewed by Tim Horton.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/GPUProcessSessionParameters.h:
(WebKit::GPUProcessSessionParameters::encode const): Deleted.
(WebKit::GPUProcessSessionParameters::decode): Deleted.
* Source/WebKit/GPUProcess/GPUProcessSessionParameters.serialization.in: Added.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.h:
(WebKit::RemoteRenderingBackendCreationParameters::encode const): Deleted.
(WebKit::RemoteRenderingBackendCreationParameters::decode): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.serialization.in: Added.
* Source/WebKit/GPUProcess/media/InitializationSegmentInfo.h:
(WebKit::InitializationSegmentInfo::TrackInformation::encode const): Deleted.
(WebKit::InitializationSegmentInfo::TrackInformation::decode): Deleted.
(WebKit::InitializationSegmentInfo::encode const): Deleted.
(WebKit::InitializationSegmentInfo::decode): Deleted.
* Source/WebKit/GPUProcess/media/InitializationSegmentInfo.serialization.in: Added.
* Source/WebKit/GPUProcess/media/MediaDescriptionInfo.h: Copied from Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.h.
(WebKit::MediaDescriptionInfo::MediaDescriptionInfo):
* Source/WebKit/GPUProcess/media/MediaDescriptionInfo.serialization.in: Added.
* Source/WebKit/GPUProcess/media/RemoteMediaDescription.h:
(WebKit::MediaDescriptionInfo::MediaDescriptionInfo): Deleted.
(WebKit::MediaDescriptionInfo::encode const): Deleted.
(WebKit::MediaDescriptionInfo::decode): Deleted.
* Source/WebKit/GPUProcess/media/TextTrackPrivateRemoteConfiguration.h:
(WebKit::TextTrackPrivateRemoteConfiguration::encode const): Deleted.
(WebKit::TextTrackPrivateRemoteConfiguration::decode): Deleted.
* Source/WebKit/GPUProcess/media/TextTrackPrivateRemoteConfiguration.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/256084@main">https://commits.webkit.org/256084@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2549e039b67dcf6548b0a0b3f0c2358b812ea63e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94584 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3749 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27473 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104218 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3810 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31943 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86897 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/100187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100251 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80959 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29763 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/84665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84227 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38345 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36204 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19322 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4201 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40106 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41971 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42069 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/38551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/GPUProcess/media/MediaDescriptionInfo.h:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    AtomString m_codec;
+    bool m_isVideo { false };
```
public m_-prefixed struct members??

achristensen07 replied:
Indeed.

Commit:
Migrate GPUProcess structures to generated serialization
https://bugs.webkit.org/show_bug.cgi?id=247150
rdar://101651187

Reviewed by Tim Horton.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/GPUProcessSessionParameters.h:
(WebKit::GPUProcessSessionParameters::encode const): Deleted.
(WebKit::GPUProcessSessionParameters::decode): Deleted.
* Source/WebKit/GPUProcess/GPUProcessSessionParameters.serialization.in: Added.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.h:
(WebKit::RemoteRenderingBackendCreationParameters::encode const): Deleted.
(WebKit::RemoteRenderingBackendCreationParameters::decode): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.serialization.in: Added.
* Source/WebKit/GPUProcess/media/InitializationSegmentInfo.h:
(WebKit::InitializationSegmentInfo::TrackInformation::encode const): Deleted.
(WebKit::InitializationSegmentInfo::TrackInformation::decode): Deleted.
(WebKit::InitializationSegmentInfo::encode const): Deleted.
(WebKit::InitializationSegmentInfo::decode): Deleted.
* Source/WebKit/GPUProcess/media/InitializationSegmentInfo.serialization.in: Added.
* Source/WebKit/GPUProcess/media/MediaDescriptionInfo.h: Copied from Source/WebKit/GPUProcess/graphics/RemoteRenderingBackendCreationParameters.h.
(WebKit::MediaDescriptionInfo::MediaDescriptionInfo):
* Source/WebKit/GPUProcess/media/MediaDescriptionInfo.serialization.in: Added.
* Source/WebKit/GPUProcess/media/RemoteMediaDescription.h:
(WebKit::MediaDescriptionInfo::MediaDescriptionInfo): Deleted.
(WebKit::MediaDescriptionInfo::encode const): Deleted.
(WebKit::MediaDescriptionInfo::decode): Deleted.
* Source/WebKit/GPUProcess/media/TextTrackPrivateRemoteConfiguration.h:
(WebKit::TextTrackPrivateRemoteConfiguration::encode const): Deleted.
(WebKit::TextTrackPrivateRemoteConfiguration::decode): Deleted.
* Source/WebKit/GPUProcess/media/TextTrackPrivateRemoteConfiguration.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/256084@main
